src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.h

```diff
@@ -75,7 +75,12 @@ static inline ucc_status_t ucc_tl_mlx5_mcast_send(ucc_tl_mlx5_mcast_coll_comm_t
         if (zcopy) {
             pp->context = (uintptr_t) PTR_OFFSET(req->ptr, offset);
         } else {
-            memcpy((void*) pp->buf, PTR_OFFSET(req->ptr, offset), length);
+            if (comm->device_mem_enabled) {
```

Comment:
is there a reason for not using `ucc_mc_memcpy` ?

Reply:
please refer to https://github.com/openucx/ucc/pull/989#discussion_r1643299598